flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/ZooKeeperStateHandleStore.java

```diff
@@ -346,11 +346,7 @@ public int exists(String pathInZooKeeper) throws Exception {
 			} else {
 				// Initial cVersion (number of changes to the children of this node)
 				int initialCVersion = stat.getCversion();
-
-				List<String> children = ZKPaths.getSortedChildren(
-						client.getZookeeperClient().getZooKeeper(),
-						ZKPaths.fixForNamespace(client.getNamespace(), "/"));
-
+				List<String> children = client.getZookeeperClient().getZooKeeper().getChildren(ZKPaths.fixForNamespace(client.getNamespace(), "/"), false);
```

Comment:
Let me do it my older way. I had a patch but I thought this is better. I checked the javadoc of the ZKPaths only. I will push my initial version of the patch only then, where convert the List<String> to List<Long> and then use that as the sorted one.